src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetGeneralTitle/index.tsx

```diff
@@ -0,0 +1,137 @@
+import { useUpdateVideo } from 'data/queries';
+import { Video } from 'types/tracks';
+import { report } from 'utils/errors/report';
+import { debounceVideo } from 'utils/widgets/widgets';
+
+const messages = defineMessages({
+  info: {
+    defaultMessage:
+      'This widget allows you to set the title of your live, alongside activate / deactivate recording of it.',
+    description:
+      'Info of the widget used for setting title and live recording.',
+    id: 'components.DashboardVideoLiveWidgetGeneralTitle.info',
+  },
+  title: {
+    defaultMessage: 'General',
+    description:
+      'Title of the widget used for setting live title and activate recording.',
+    id: 'components.DashboardVideoLiveWidgetGeneralTitle.title',
+  },
+  placeholderTitleInput: {
+    defaultMessage: 'Enter title of your live here',
+    description:
+      'A placeholder text indicating the purpose of the input and what it is supposed to received.',
+    id: 'components.DashboardVideoLiveWidgetGeneralTitle.placeholderTitleInput',
+  },
+  updateVideoSucces: {
+    defaultMessage: 'Video updated.',
+    description: 'Message displayed when video is successfully updated.',
+    id: 'component.DashboardVideoLiveWidgetGeneralTitle.updateVideoSuccess',
+  },
+  updateVideoFail: {
+    defaultMessage: 'Video update has failed !',
+    description: 'Message displayed when video update has failed.',
+    id: 'component.DashboardVideoLiveWidgetGeneralTitle.updateVideoFail',
+  },
+  updateTitleBlank: {
+    defaultMessage: "Title can't be blank !",
+    description:
+      'Message displayed when the user tried to enter a blank title.',
+    id: 'component.DashboardVideoLiveWidgetGeneralTitle.updateTitleBlank',
+  },
+  recordingToggleLabel: {
+    defaultMessage: 'Activate live recording',
+    description:
+      'The label associated to the toggle button reponsible of live recording activation / deactivation.',
+    id: 'components.DashboardVideoLiveWidgetGeneralTitle.liveRecordingToggleLabel',
+  },
+});
+
+interface DashboardVideoLiveWidgetGeneralTitleProps {
+  video: Video;
+}
+
+export const DashboardVideoLiveWidgetGeneralTitle = ({
+  video,
+}: DashboardVideoLiveWidgetGeneralTitleProps) => {
+  const intl = useIntl();
+  const [title, setTitle] = useState(video.title);
+  const [checked, setChecked] = useState(video.allow_recording);
+
+  const videoMutation = useUpdateVideo(video.id, {
+    onSuccess: () => {
+      toast.success(intl.formatMessage(messages.updateVideoSucces), {
+        position: 'bottom-center',
+      });
+    },
+    onError: (err, variables) => {
+      if ('title' in variables) {
+        setTitle(video.title);
+      }
+      if ('allow_recording' in variables) {
+        setChecked(video.allow_recording);
+      }
+      report(err);
+      toast.error(intl.formatMessage(messages.updateVideoFail), {
+        position: 'bottom-center',
+      });
+    },
+  });
+
+  const debouncedUpdatedVideo = useCallback(
```

Comment:
I'm not fan of this usage. I think it's error prone if the dependency array is not used as expected. IMO using `useRef` is safer

Reply:
In this case, I use a useCallback because of the following if statement :   
`      if (updatedVideoProperty.title === '') {
        toast.error(intl.formatMessage(messages.updateTitleBlank), {
          position: 'bottom-center',
        });
        setTitle(video.title);
        return;
      }`
Indeed, if the instructor tries to enter an empty title, this if statement will prevent it, display a toast and fill the now empty input with the value `video.title`. But for this behavior to work, we need it to be updated and be given in a deps array. IMO this behavior won't be keep with a useRef.